### Assignments

To close out this checkpoint, you're going to do three assignments. For the first assignment, you'll write up a short answer to a question. For the second two assignments, you'll do your work in Jupyter notebooks.

Please submit links to all your work below. This is not a graded checkpoint, but you should discuss your solutions with your mentor. Also, when you're done, compare your work to these example solutions.

1. Interpretation and significance

Suppose that we would like to know how much families in the US are spending on recreation annually. We've estimated the following model:

𝑒𝑥𝑝𝑒𝑛𝑑𝑖𝑡𝑢𝑟𝑒 = 873 + 0.0012𝑎𝑛𝑛𝑢𝑎𝑙_𝑖𝑛𝑐𝑜𝑚𝑒 + 0.00002𝑎𝑛𝑛𝑢𝑎𝑙_𝑖𝑛𝑐𝑜𝑚𝑒2 − 223.57ℎ𝑎𝑣𝑒_𝑘𝑖𝑑𝑠

expenditure is the annual spending on recreation in US dollars, annual_income is the annual income in US dollars, and have_kids is a dummy variable indicating the families with children. Interpret the estimated coefficients. What additional statistics should be given in order to make sure that your interpretations make sense statistically. Write up your answer.

2. Weather model

In this exercise, you'll work with the historical temperature data from the previous checkpoint. To complete this assignment, submit a link to a Jupyter notebook containing your solutions to the following tasks:

    First, load the dataset from the weatherinszeged table from Thinkful's database.
    Build a linear regression model where your target variable is the difference between the apparenttemperature and the temperature. As explanatory variables, use humidity and windspeed. Now, estimate your model using OLS. Are the estimated coefficients statistically significant? Are the signs of the estimated coefficients in line with your previous expectations? Interpret the estimated coefficients. What are the relations between the target and the explanatory variables?
    Next, include the interaction of humidity and windspeed to the model above and estimate the model using OLS. Are the coefficients statistically significant? Did the signs of the estimated coefficients for humidity and windspeed change? Interpret the estimated coefficients.

3. House prices model

In this exercise, you'll interpret your house prices model. To complete this assignment, submit a link to a Jupyter notebook containing your solutions to the following tasks:

    Load the houseprices data from Thinkful's database.
    Run your house prices model again and interpret the results. Which features are statistically significant, and which are not?
    Now, exclude the insignificant features from your model. Did anything change?
    Interpret the statistically significant coefficients by quantifying their relations with the house prices. Which features have a more prominent effect on house prices?
    Do the results sound reasonable to you? If not, try to explain the potential reasons.



#### Part 1:  Interpretation and significance

Suppose that we would like to know how much families in the US are spending on recreation annually. We've estimated the following model:

𝑒𝑥𝑝𝑒𝑛𝑑𝑖𝑡𝑢𝑟𝑒 = 873 + 0.0012𝑎𝑛𝑛𝑢𝑎𝑙_𝑖𝑛𝑐𝑜𝑚𝑒 + 0.00002𝑎𝑛𝑛𝑢𝑎𝑙_𝑖𝑛𝑐𝑜𝑚𝑒2 − 223.57ℎ𝑎𝑣𝑒_𝑘𝑖𝑑𝑠

expenditure is the annual spending on recreation in US dollars, annual_income is the annual income in US dollars, and have_kids is a dummy variable indicating the families with children. Interpret the estimated coefficients. What additional statistics should be given in order to make sure that your interpretations make sense statistically. Write up your answer.

𝑦=𝛽0+𝛽1𝑥1+𝛽2𝑥2+...+𝛽𝑛𝑥𝑛+𝜖(𝑒𝑞.1)

target = intercept + features (coefficiencts)+ error (residual)


Y (expenditure) = intercept (+873) + 0.0012𝑎𝑛𝑛𝑢𝑎𝑙_𝑖𝑛𝑐𝑜𝑚𝑒 (coef) + 0.00002𝑎𝑛𝑛𝑢𝑎𝑙_𝑖𝑛𝑐𝑜𝑚𝑒2 (coef) − 223.57ℎ𝑎𝑣𝑒_𝑘𝑖𝑑𝑠 (coef) + error (unexplained feature)

Need to know stastical significance of the features (coef) with t-test & p-value. Specifically, are the coefficients different from zero? It appears that they are, but a t-test is required to show that they are *stastically significant*. 
For purpose of interpreting the model, we will assume that the coefficients are stastically significant.

The intercept or bias of the model is 873. The coefficients 'annual_income' and 'annual_income2' have a positive relationship with the target. The 'have_kids' coefficient has a negative relationship with the targer: these families will spend -$223.57 less than families without kids. 

In [1]:
#from answer
import numpy as np
import matplotlib.pyplot as plt

# X represents annual income
X = np.arange(10000, 50000, 1000)
# Y represents annual expenditure on recreation
# for families with no children
Y = 873 + 0.0012*X + 0.000002*(X**2)
# Y_kids represents annual expenditure on recreation
# for families with children
Y_kids = (873 - 223.57) + 0.0012*X + 0.000002*(X**2)


plt1 = plt.plot(X, Y, label="with no children")
plt2 = plt.plot(X, Y_kids, label="with children")
plt.xlabel("income")
plt.ylabel("recreation expenditure")
plt.title("The relation between income and recreation expenditure")
plt.legend()
plt.show()

<Figure size 640x480 with 1 Axes>

#### Part 2:  Weather model

In this exercise, you'll work with the historical temperature data from the previous checkpoint. To complete this assignment, submit a link to a Jupyter notebook containing your solutions to the following tasks:

1. First, load the dataset from the weatherinszeged table from Thinkful's database.
2. Build a linear regression model where your target variable is the difference between the apparenttemperature and the temperature. As explanatory variables, use humidity and windspeed. Now, estimate your model using OLS. Are the estimated coefficients statistically significant? Are the signs of the estimated coefficients in line with your previous expectations? Interpret the estimated coefficients. What are the relations between the target and the explanatory variables?
3. Next, include the interaction of humidity and windspeed to the model above and estimate the model using OLS. Are the coefficients statistically significant? Did the signs of the estimated coefficients for humidity and windspeed change? Interpret the estimated coefficients.

##### First, load the dataset from the weatherinszeged table from Thinkful's database.

In [2]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from sklearn import linear_model
import statsmodels.formula.api as smf
from sqlalchemy import create_engine

# Display preferences.
%matplotlib inline
pd.options.display.float_format = '{:.3f}'.format

import warnings
warnings.filterwarnings(action="ignore")

postgres_user = 'dsbc_student'
postgres_pw = '7*.8G9QH21'
postgres_host = '142.93.121.174'
postgres_port = '5432'
postgres_db = 'weatherinszeged'

engine = create_engine('postgresql://{}:{}@{}:{}/{}'.format(
    postgres_user, postgres_pw, postgres_host, postgres_port, postgres_db))

weather_df = pd.read_sql_query('select * from weatherinszeged',con=engine)

# no need for an open connection, as we're only doing a single query
engine.dispose()


#####  Build a linear regression model where your target variable is the difference between the apparenttemperature and the temperature. 
As explanatory variables, use **humidity and windspeed.** 
Now, **estimate your model using OLS**. Are the estimated coefficients statistically significant? Are the signs of the estimated coefficients in line with your previous expectations? Interpret the estimated coefficients. What are the relations between the target and the explanatory variables?

In [3]:
weather_df.head()

date        summary preciptype  temperature  \
0 2006-03-31 22:00:00+00:00  Partly Cloudy       rain        9.472   
1 2006-03-31 23:00:00+00:00  Partly Cloudy       rain        9.356   
2 2006-04-01 00:00:00+00:00  Mostly Cloudy       rain        9.378   
3 2006-04-01 01:00:00+00:00  Partly Cloudy       rain        8.289   
4 2006-04-01 02:00:00+00:00  Mostly Cloudy       rain        8.756   

   apparenttemperature  humidity  windspeed  windbearing  visibility  \
0                7.389     0.890     14.120      251.000      15.826   
1                7.228     0.860     14.265      259.000      15.826   
2                9.378     0.890      3.928      204.000      14.957   
3                5.944     0.830     14.104      269.000      15.826   
4                6.978     0.830     11.045      259.000      15.826   

   loudcover  pressure                       dailysummary  
0      0.000  1015.130  Partly cloudy throughout the day.  
1      0.000  1015.630  Partly cloudy throughout the day.  
2      0.000  1015.940  Partly cloudy throughout the day.  
3      0.000  1016.410  Partly cloudy throughout the day.  
4      0.000  1016.510  Partly cloudy throughout the day.

In [4]:
weather_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96453 entries, 0 to 96452
Data columns (total 12 columns):
date                   96453 non-null datetime64[ns, UTC]
summary                96453 non-null object
preciptype             96453 non-null object
temperature            96453 non-null float64
apparenttemperature    96453 non-null float64
humidity               96453 non-null float64
windspeed              96453 non-null float64
windbearing            96453 non-null float64
visibility             96453 non-null float64
loudcover              96453 non-null float64
pressure               96453 non-null float64
dailysummary           96453 non-null object
dtypes: datetime64[ns, UTC](1), float64(8), object(3)
memory usage: 8.8+ MB


In [5]:
weather_df.describe()

temperature  apparenttemperature  humidity  windspeed  windbearing  \
count    96453.000            96453.000 96453.000  96453.000    96453.000   
mean        11.933               10.855     0.735     10.811      187.509   
std          9.552               10.697     0.195      6.914      107.383   
min        -21.822              -27.717     0.000      0.000        0.000   
25%          4.689                2.311     0.600      5.828      116.000   
50%         12.000               12.000     0.780      9.966      180.000   
75%         18.839               18.839     0.890     14.136      290.000   
max         39.906               39.344     1.000     63.853      359.000   

       visibility  loudcover  pressure  
count   96453.000  96453.000 96453.000  
mean       10.347      0.000  1003.236  
std         4.192      0.000   116.970  
min         0.000      0.000     0.000  
25%         8.340      0.000  1011.900  
50%        10.046      0.000  1016.450  
75%        14.812      0.000  1021.090  
max        16.100      0.000  1046.380

In [6]:
weather_df.isnull().sum()

date                   0
summary                0
preciptype             0
temperature            0
apparenttemperature    0
humidity               0
windspeed              0
windbearing            0
visibility             0
loudcover              0
pressure               0
dailysummary           0
dtype: int64

Build a linear regression model where your **target variable is the difference between the apparenttemperature and the temperature.** 

In [7]:
Y = weather_df['apparenttemperature'] - weather_df['temperature']
# X is the feature set which includes
X = weather_df[['humidity','windspeed']]

import statsmodels.api as sm
X = sm.add_constant(X)

results = sm.OLS(Y, X).fit()

results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Tue, 22 Oct 2019   Prob (F-statistic):               0.00
Time:                        14:50:39   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.4381      0.021    115.948      0.000       2.397       2.479
humidity      -3.0292      0.024   -126.479      0.000      -3.076      -2.982
windspeed     -0.1193      0.001   -176.164      0.000      -0.121      -0.118
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.267
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                          -0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The t-test and associated p-value help to quantify if/how much an individual feature (coefficient) has impacted the variance in the target feature. The null hypothesis would be that each variable (coefficient) does NOT impact the outcome (target). A lower p-value allows us to reject the null hypothesis, meaning the variable in question DID have an impact on the outcome (target). 

**Are the estimated coefficients statistically significant?**
Yes, the p-value for both coefficients is <0.05, which indicates that they are stastically significant. 

**Are the signs of the estimated coefficients in line with your previous expectations?**
I would expect temperature and humidity to increase and decrease in the same direction, as humidity holds heat in the air. However, the target is the difference between apparent temperature and temperature. In the summer, the apparent temperature can seem higer than the actual temperature and vice versa in the winter. 
For wind speed, I would expect that it would lower the apparent temperature in relation to the actual temperature, in both summer and winter, so the negative correlation makes sense. 

**Interpret the estimated coefficients. What are the relations between the target and the explanatory variables?**
As the humidity and windspeed rise by one point, they lower the target (difference in apparent and actual temperature) by 3.03 and 0.13 degrees, respectively. 

#####  Next, include the interaction of humidity and windspeed to the model above and estimate the model using OLS. 
Are the coefficients statistically significant? Did the signs of the estimated coefficients for humidity and windspeed change? Interpret the estimated coefficients.

In [8]:
weather_df['humd_wind_intact'] = weather_df['humidity'] * weather_df['windspeed']

In [9]:

Y = weather_df['apparenttemperature'] - weather_df['temperature']
# X is the feature set which includes
X = weather_df[['humidity','windspeed', 'humd_wind_intact']]

import statsmodels.api as sm
X = sm.add_constant(X)

results = sm.OLS(Y, X).fit()

results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Tue, 22 Oct 2019   Prob (F-statistic):               0.00
Time:                        14:50:39   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const                0.0839      0.033      2.511      0.012       0.018       0.149
humidity             0.1775      0.043      4.133      0.000       0.093       0.262
windspeed            0.0905      0.002     36.797      0.000       0.086       0.095
humd_wind_intact    -0.2971      0.003    -88.470      0.000      -0.304      -0.291
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.265
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                          -0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

**Are the coefficients statistically significant?**

Yes, the previous coefficients remain stastically significant, as does the new coefficient. 

**Did the signs of the estimated coefficients for humidity and windspeed change? Interpret the estimated coefficients.**

The signs for the coefficients did change: 
original humidity = -3.0292, revised humidity = 0.1775; 
original windspeed = -0.1193, revised windspeed = 0.0905.
Originally these coefficients were negatively correlated, but adding the interaction between them changed both the sign and impact on the outcome. Humidity now will increase the target by 0.18 degrees for every one point increase, and wind speed will increase the target by 0.09 degrees for every one point increase in the coefficient. 


#### Part 3: House prices model

In this exercise, you'll interpret your house prices model. To complete this assignment, submit a link to a Jupyter notebook containing your solutions to the following tasks:

1. Load the houseprices data from Thinkful's database.
2. Run your house prices model again and interpret the results. Which features are statistically significant, and which are not?
3. Now, exclude the insignificant features from your model. Did anything change?
4. Interpret the statistically significant coefficients by quantifying their relations with the house prices. Which features have a more prominent effect on house prices?
5. Do the results sound reasonable to you? If not, try to explain the potential reasons.

##### Load the houseprices data from Thinkful's database

In [10]:
import numpy as np
import pandas as pd
from sklearn import linear_model
import matplotlib.pyplot as plt
from sqlalchemy import create_engine

import warnings
warnings.filterwarnings('ignore')

postgres_user = 'dsbc_student'
postgres_pw = '7*.8G9QH21'
postgres_host = '142.93.121.174'
postgres_port = '5432'
postgres_db = 'houseprices'

In [11]:
engine = create_engine('postgresql://{}:{}@{}:{}/{}'.format(
    postgres_user, postgres_pw, postgres_host, postgres_port, postgres_db))
houseprices_df = pd.read_sql_query('select * from houseprices',con=engine)

# no need for an open connection, as we're only doing a single query
engine.dispose()

In [12]:
houseprices_df['central_air_ohc'] = pd.get_dummies(houseprices_df.centralair, drop_first=True)

mszoning_ohc_df = pd.get_dummies(houseprices_df.mszoning, prefix = 'mszoning', drop_first=True)
houseprices_df = pd.concat([houseprices_df, mszoning_ohc_df], axis=1)

kitchenqual_ohc_df = pd.get_dummies(houseprices_df.kitchenqual, prefix = 'kitchenqual',drop_first=True)
houseprices_df = pd.concat([houseprices_df, kitchenqual_ohc_df], axis=1)

fullbath_ohc_df = pd.get_dummies(houseprices_df.fullbath, prefix = 'fullbath', drop_first=True)
houseprices_df = pd.concat([houseprices_df, fullbath_ohc_df], axis=1)

In [13]:
numerics_df = houseprices_df.select_dtypes(include=[np.number])

In [14]:
np.abs(numerics_df.iloc[:,1:].corr().loc[:,'saleprice']).sort_values(ascending=False)

saleprice         1.000
overallqual       0.791
grlivarea         0.709
garagecars        0.640
garagearea        0.623
totalbsmtsf       0.614
firstflrsf        0.606
fullbath          0.561
totrmsabvgrd      0.534
yearbuilt         0.523
fullbath_1        0.521
kitchenqual_TA    0.519
yearremodadd      0.507
garageyrblt       0.486
masvnrarea        0.477
fireplaces        0.467
fullbath_2        0.426
bsmtfinsf1        0.386
lotfrontage       0.352
wooddecksf        0.324
kitchenqual_Gd    0.322
fullbath_3        0.320
secondflrsf       0.319
openporchsf       0.316
mszoning_RM       0.288
halfbath          0.284
lotarea           0.264
central_air_ohc   0.251
mszoning_RL       0.245
bsmtfullbath      0.227
bsmtunfsf         0.214
bedroomabvgr      0.168
kitchenqual_Fa    0.157
kitchenabvgr      0.136
enclosedporch     0.129
screenporch       0.111
poolarea          0.092
mszoning_FV       0.090
mssubclass        0.084
overallcond       0.078
mszoning_RH       0.065
mosold          

Choosing only variables with corr >0.5:
overallqual        0.790982,
grlivarea          0.708624,
garagecars         0.640409,
garagearea         0.623431,
totalbsmtsf        0.613581,
firstflrsf         0.605852,
fullbath           0.560664,
totrmsabvgrd       0.533723,
yearbuilt          0.522897,
fullbath_1         0.520796,
kitchenqual_TA     0.519298,
yearremodadd       0.507101

In [15]:
features = ['overallqual', 'grlivarea', 'garagecars',
            'garagearea', 'totalbsmtsf', 'firstflrsf', 
            'fullbath', 'totrmsabvgrd', 'yearbuilt',
            'fullbath_1', 'kitchenqual_TA', 'yearremodadd']

##### Run your house prices model again and interpret the results. Which features are statistically significant, and which are not?

In [16]:
import statsmodels.api as sm

Y = houseprices_df['saleprice']

X = houseprices_df[['overallqual', 'grlivarea', 'garagecars',
            'garagearea', 'totalbsmtsf', 'firstflrsf', 
            'fullbath', 'totrmsabvgrd', 'yearbuilt',
            'fullbath_1', 'kitchenqual_TA', 'yearremodadd']]

results = sm.OLS(Y, X).fit()

results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              saleprice   R-squared:                       0.963
Model:                            OLS   Adj. R-squared:                  0.962
Method:                 Least Squares   F-statistic:                     3121.
Date:                Tue, 22 Oct 2019   Prob (F-statistic):               0.00
Time:                        14:50:40   Log-Likelihood:                -17472.
No. Observations:                1460   AIC:                         3.497e+04
Df Residuals:                    1448   BIC:                         3.503e+04
Df Model:                          12                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
overallqual     2.125e+04   1179.331     18.016      0.000    1.89e+04    2.36e+04
grlivarea         44.4716      4.244     10.480      0.000      36.147      52.796
garagecars      1.434e+04   3065.587      4.677      0.000    8323.571    2.04e+04
garagearea        11.9567     10.426      1.147      0.252      -8.495      32.408
totalbsmtsf       22.2268      4.320      5.145      0.000      13.753      30.700
firstflrsf        12.8085      4.970      2.577      0.010       3.060      22.557
fullbath        1.389e+04   4912.915      2.828      0.005    4254.880    2.35e+04
totrmsabvgrd      -8.9566   1133.766     -0.008      0.994   -2232.956    2215.043
yearbuilt        126.9411     46.524      2.729      0.006      35.680     218.203
fullbath_1      1.986e+04   5213.281      3.809      0.000    9632.798    3.01e+04
kitchenqual_TA -1.736e+04   2485.884     -6.983      0.000   -2.22e+04   -1.25e+04
yearremodadd    -180.0707     46.179     -3.899      0.000    -270.656     -89.486
==============================================================================
Omnibus:                      442.552   Durbin-Watson:                   1.977
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            40022.268
Skew:                          -0.374   Prob(JB):                         0.00
Kurtosis:                      28.639   Cond. No.                     2.48e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.48e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

**Which features are statistically significant, and which are not?**

Significant: 'overallqual', 'grlivarea', 'garagecars',
            'totalbsmtsf', 'firstflrsf', 
            'fullbath', 'yearbuilt',
            'fullbath_1', 'kitchenqual_TA', 'yearremodadd'
            
Not significant: 'garagearea', 'totrmsabvgrd', 

##### Now, exclude the insignificant features from your model. Did anything change?

In [17]:
Y = houseprices_df['saleprice']

X = houseprices_df[['overallqual', 'grlivarea', 'garagecars',
            'totalbsmtsf', 'firstflrsf', 'fullbath', 'yearbuilt',
            'fullbath_1', 'kitchenqual_TA', 'yearremodadd']]

results = sm.OLS(Y, X).fit()

results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              saleprice   R-squared:                       0.963
Model:                            OLS   Adj. R-squared:                  0.962
Method:                 Least Squares   F-statistic:                     3747.
Date:                Tue, 22 Oct 2019   Prob (F-statistic):               0.00
Time:                        14:50:41   Log-Likelihood:                -17473.
No. Observations:                1460   AIC:                         3.497e+04
Df Residuals:                    1450   BIC:                         3.502e+04
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
overallqual     2.123e+04   1177.093     18.036      0.000    1.89e+04    2.35e+04
grlivarea         44.7407      3.017     14.829      0.000      38.822      50.659
garagecars      1.717e+04   1804.843      9.515      0.000    1.36e+04    2.07e+04
totalbsmtsf       22.6440      4.291      5.277      0.000      14.226      31.061
firstflrsf        13.2023      4.957      2.664      0.008       3.479      22.925
fullbath        1.384e+04   4903.414      2.822      0.005    4218.682    2.35e+04
yearbuilt        128.5306     46.416      2.769      0.006      37.480     219.581
fullbath_1       2.02e+04   5203.250      3.882      0.000    9993.598    3.04e+04
kitchenqual_TA -1.746e+04   2460.242     -7.096      0.000   -2.23e+04   -1.26e+04
yearremodadd    -181.9672     45.954     -3.960      0.000    -272.111     -91.824
==============================================================================
Omnibus:                      431.663   Durbin-Watson:                   1.976
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            37480.171
Skew:                          -0.336   Prob(JB):                         0.00
Kurtosis:                      27.813   Cond. No.                     2.46e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.46e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

**Did anything change?**
The 'firstfl' variable became slightly more significant. 
The value of the coefficients has fluctuated, but not more than +/- few dollars. 

##### Interpret the statistically significant coefficients by quantifying their relations with the house prices. Which features have a more prominent effect on house prices?
For every point increase in the coefficient, the cost of the house increases or decreases as follows: 

overallqual: 21,230.00,  fullbath_1: 20,200.00,  kitchenqual_TA: -17,460.00,  garagecars: 17,170.00,  fullbath: 13,840.00,  yearremodadd: -181.97,  
yearbuilt: 128.53,  grlivarea: 44.74,  
totalbsmtsf: 22.64,  firstflrsf: 13.20 

Clearly, the biggest impact of the cost of the house is the overall quality, followed by at least 1 full bath, (kitchen quality), the number of cars that fit into the garage, and the general full bath category. 
Having a condition that is in typical/average condition will REDUCE the house cost by approximately $17,500. 
Also, the year remodeled also has a negative impact on sale price. The other coefficients have a much smaller impact on the sales price. 

##### Do the results sound reasonable to you? If not, try to explain the potential reasons.

The results do seem reasonable to me, with a couple exceptions. The prevailing wisom (at least as I know it to be) is that the most important feature impacting the value of a house is its location. I am genuinely surprised that the neighborhood is not correlated with the sale price. Perhaps it is due to the geographical location of the data: there may not be much difference in neighborhoods (schools, taxes, etc) or there is no perceived difference related to social status. In the Philadelphia area, there is a huge difference in house prices based on zip code, with homes literally one street apart. 

The fact that having only an average/typical kitchen will DECREASE the value of the house does not surprise me. The second bit of real estate wisdom I've heard is that improving your kitchen and/or bathroom increases the value of your house, everything else just makes you happy. Home buyers generally expect a house not simply to be "move-in" ready, but are looking for an updated kitchen.   